astroquery/mast/core.py

```diff
@@ -379,458 +96,47 @@ def session_info(self, silent=None, verbose=None):
-        if not all_results:
-            warnings.warn("Query returned no results.", NoResultsWarning)
-        return all_results
-
-    def logout(self):  # pragma: no cover
+    def logout(self):
         """
         Log out of current MAST session.
         """
-        self._session.cookies.clear_session_cookies()
+        self._auth_obj.logout()
         self._authenticated = False
 
-    @class_or_instance
-    def service_request_async(self, service, params, pagesize=None, page=None, **kwargs):
-        """
-        Given a Mashup service and parameters, builds and excecutes a Mashup query.
-        See documentation `here <https://mast.stsci.edu/api/v0/class_mashup_1_1_mashup_request.html>`__
-        for information about how to build a Mashup request.
-
-        Parameters
-        ----------
-        service : str
-            The Mashup service to query.
-        params : dict
-            JSON object containing service parameters.
-        pagesize : int, optional
-            Default None.
-            Can be used to override the default pagesize (set in configs) for this query only.
-            E.g. when using a slow internet connection.
-        page : int, optional
-            Default None.
-            Can be used to override the default behavior of all results being returned to obtain
-            a specific page of results.
-        **kwargs :
-            See MashupRequest properties
-            `here <https://mast.stsci.edu/api/v0/class_mashup_1_1_mashup_request.html>`__
-            for additional keyword arguments.
-
-        Returns
-        -------
-        response : list of `~requests.Response`
-        """
-
-        # setting self._current_service
-        if service not in self._column_configs.keys():
-            fetch_name = kwargs.pop('fetch_name', None)
-            self._get_col_config(service, fetch_name)
-        self._current_service = service
-
-        # setting up pagination
-        if not pagesize:
-            pagesize = self.PAGESIZE
-        if not page:
-            page = 1
-            retrieve_all = True
-        else:
-            retrieve_all = False
-
-        headers = {"User-Agent": self._session.headers["User-Agent"],
-                   "Content-type": "application/x-www-form-urlencoded",
-                   "Accept": "text/plain"}
-
-        mashup_request = {'service': service,
-                         'params': params,
-                         'format': 'json',
-                         'pagesize': pagesize,
-                         'page': page}
-
-        for prop, value in kwargs.items():
-            mashup_request[prop] = value
-
-        req_string = _prepare_service_request_string(mashup_request)
-        response = self._request("POST", self._MAST_REQUEST_URL, data=req_string, headers=headers,
-                                 retrieve_all=retrieve_all)
-
-        return response
+    @deprecated(since="v0.3.9", alternative="enable_cloud_dataset")
```

Comment:
This may be a good time to remove this, too.

Reply:
(I won't repeat this comment, but there are other deprecations, so consider removing those methods, too.